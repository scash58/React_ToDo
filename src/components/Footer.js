import React from 'react';
import Logout from './Auth/Logout';
import { useAuth } from '../contexts/AuthContext';

export default function Footer() {
    const {currentUser} = useAuth();

  return (
    <>
      {currentUser &&
        <Logout />
      }
      <footer className="text-center text-white bg-dark p-4">
          <p className="bold">
              <em>&copy; {new Date().getFullYear()} All rights reserved</em>
          </p>
      </footer>
    </>
  );
}
