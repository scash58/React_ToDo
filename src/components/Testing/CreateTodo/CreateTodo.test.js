import React from "react";
import ReactDOM from "react-dom";
import CreateTodo from "./CreateTodo";
import Footer from "./Footer"

//The below items are imported to bring in specific testing functionalities. First are some tools that allow for shallow rendering and cleanup of data after each test. Second are tools that actually make these test function.
import {render, cleanup} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
//npm install react-test-renderer - rendering fuctionality ../contexts/AuthContext ./Auth/Logout
import renderer from 'react-test-renderer'
import EndToEnd from '../EndToEnd'
import ShallowRender from "../ShallowRender";
import reactDom from "react-dom";

afterEach(cleanup);

it('CreateTodo - matches snapshot', () => {
    const tree = renderer.create(<CreateTodo />).toJSON();

    expect(tree).toMatchSnapshot();
})

it('Footer - renders without crashing', () => {
    const div = document.createElement('div');
    reactDom.render(<Footer></Footer>, div);
})
