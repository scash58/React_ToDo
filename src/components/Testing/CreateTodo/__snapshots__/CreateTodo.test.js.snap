// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTodo - matches snapshot 1`] = `
<article
  className="createTodo m-2 text-white justify-content-center"
>
  <form
    action="#"
    id="todoForm"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <div
      className="form-group m-3"
    >
      <input
        className="form-control"
        name="Action"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="Task to be compleated"
        value=""
      />
    </div>
    <div
      className="form-group m-3"
    >
      <select
        className="form-control"
        name="CategoryId"
        onBlur={[Function]}
        onChange={[Function]}
        value=""
      >
        <option
          disabled={true}
          value="0"
        >
          [--Please Choose--]
        </option>
      </select>
    </div>
    <div
      className="form-group m-3"
    >
      <button
        className="btn btn-info m-3"
        type="submit"
      >
        Submit Todo
      </button>
    </div>
  </form>
</article>
`;
